modules/geo-location/class.jetpack-geo-location.php

```diff
@@ -0,0 +1,432 @@
+
+		return $html;
+	}
+
+	/**
+	 * Get the ID of the current global post object, if available.  Otherwise, return null.
+	 *
+	 * This isolates the access of the global scope to this single method, making it easier to
+	 * safeguard against unexpected missing $post objects in other hook functions.
+	 *
+	 * @return int|null
+	 */
+	public function get_post_id() {
+		global $post;
+
+		if ( ! isset( $post) || ! $post || ! is_object( $post ) || ! isset( $post->ID ) ) {
+			return null;
+		}
+
+		return $post->ID;
+	}
+
+	/**
+	 * This method always returns an array with the following structure:
+	 *
+	 * array(is_public => bool, latitude => float, longitude => float, label => string, is_populated => bool)
+	 *
+	 * So, regardless of whether your post actually has values in postmeta for the geo-location fields,
+	 * you can be sure that you can reference those array keys in calling code without having to juggle
+	 * isset(), array_key_exists(), etc.
+	 *
+	 * Mocking this method during testing can also be useful for testing output and logic in various
+	 * hook functions.
+	 *
+	 * @param integer $post_id
+	 *
+	 * @return array A predictably structured array representing the meta values for the supplied post ID.
+	 */
+	public function get_meta_values( $post_id ) {
+		$meta_values = array(
+			'is_public'    => (bool) $this->sanitize_public( $this->get_meta_value( $post_id, 'public' ) ),
+			'latitude'     => $this->sanitize_coordinate( $this->get_meta_value( $post_id, 'latitude' ) ),
+			'longitude'    => $this->sanitize_coordinate( $this->get_meta_value( $post_id, 'longitude' ) ),
+			'label'        => trim( $this->get_meta_value( $post_id, 'address' ) ),
+			'is_populated' => false,
+		);
+
+		if ( $meta_values['latitude'] && $meta_values['longitude'] && $meta_values['label'] ) {
+			$meta_values['is_populated'] = true;
+		}
+
+		return $meta_values;
+	}
+
+	/**
+	 * This function wraps get_post_meta() to enable us to keep the "geo_" prefix isolated to a single
+	 * location in the code and to assist in mocking during testing.
+	 *
+	 * @param integer $post_id
+	 * @param string $meta_field_name
+	 *
+	 * @return mixed
+	 */
+	public function get_meta_value( $post_id, $meta_field_name ) {
+		if ( ! $post_id ) {
+			return null;
+		}
+
+		return get_post_meta( $post_id, 'geo_' . $meta_field_name, true );
+	}
+
+	/**
+	 * Simple wrapper for testing purposes.
+	 *
+	 * @return bool
+	 */
+	public function is_single() {
+		return is_single();
+	}
+
+	/**
+	 * Simple wrapper for testing purposes.
+	 *
+	 * @return bool
+	 */
+	public function is_feed() {
+		return is_feed();
+	}
+
+	/**
+	 * Simple wrapper for testing purposes.
+	 *
+	 * @param string $feature
+	 * @return bool
+	 */
+	public function current_theme_supports( $feature ) {
+		return current_theme_supports( $feature );
+	}
+
+	/**
+	 * Check to see if the current filter is the get_the_excerpt filter.
+	 *
+	 * @return bool
+	 */
+	public function is_currently_excerpt_filter() {
```

Comment:
Would `current_filter()` work here?

Reply:
It actually doesn't.  Added some comments explaining why in 76922f8.